src/lsdb/catalog/catalog.py

```diff
@@ -261,6 +263,31 @@ def cone_search(self, ra: float, dec: float, radius: float):
         ddf_partition_map = {pixel: i for i, pixel in enumerate(pixels_in_cone)}
         return Catalog(cone_search_ddf, ddf_partition_map, filtered_hc_structure)
 
+    def polygon_search(self, polygon: SingleSphericalPolygon) -> Catalog:
+        """Perform a polygonal search to filter the catalog.
+
+        Filters to points within the polygonal region specified in ra and dec, in degrees.
+        Filters partitions in the catalog to those that have some overlap with the region.
+
+        Args:
+            polygon (SingleSphericalPolygon): The polygon to filter pixels with.
+
+        Returns:
+            A new catalog containing the points filtered to those within the
+            polygonal region, and the partitions that have some overlap with it.
+        """
+        filtered_hc_structure = self.hc_structure.filter_by_polygon(polygon)
+        pixels_in_polygon = filtered_hc_structure.get_healpix_pixels()
+        partitions = self.to_delayed()
+        partitions_in_polygon = [partitions[self._ddf_pixel_map[hp_pixel]] for hp_pixel in pixels_in_polygon]
+        filtered_partitions = [
+            polygon_filter(partition, polygon, self.hc_structure) for partition in partitions_in_polygon
+        ]
+        polygon_search_ddf = dd.from_delayed(filtered_partitions, meta=self._ddf._meta)
```

Comment:
I've left this so late that we can now add divisions in here.

Reply:
Nice catch! I also added them to the catalogs resulting from cone searches.